CONTRIBUTING.md

```diff
@@ -20,6 +21,46 @@ More resources can be found in the nascent [developer documentation](devdocs/REA
 - [Platform support policy](devdocs/PLATFORM_SUPPORT.md)
 - [Our Rust style](devdocs/rust_style.md)
 
+## Tips for submitting PRs
+
+Please talk to the core team before making major changes! See the [proposing design changes](#proposing-design-changes) for more details.
+
+### Linking and mentioning issues
+
+If your PR closes one or more issues, you can automatically link the PR with them by using one of the [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword):
+
+- This PR should close #xxxx
+- Fixes #xxxx
+
+You can also mention related issues, PRs or discussions!
+
+### Release notes section
+
+In our PR template, we have a "Release notes summary" section which will be included in our release notes for our blog.
+
+This section should include any information about your change which is relevant to a user of Nushell. You should try to keep it brief and simple to understand, and focus on the ways your change directly impacts the user experience. We highly encourage adding examples and, when relevant, screenshots in this section.
```

Comment:
I would add a sentence or two that this has two angles:
- intended changes for the users in additions or deliberate breaking changes
- reflecting on side effects that will also affect users (take it as a reminder to think about this and document it nevertheless even if it is a side show to other changes)